pkg/operator/controller/route-metrics/metrics_test.go

```diff
@@ -0,0 +1,67 @@
+package routemetrics
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+func TestRouteMetricsControllerRoutesPerShardMetric(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		shardName             string
+		actions               []string
+		metricValues          []float64
+		expectedMetricFormats []string
+	}{
+		{
+			name:         "routes per shard metrics test shard",
+			shardName:    "test",
+			actions:      []string{"Set", "Set", "Set", "Delete"},
+			metricValues: []float64{0, 2, 1},
+			expectedMetricFormats: []string{`
```

Comment:
Shouldn't there be 4 expectedMetricFormats, to match the 4 actions?

Reply:
Ok, thanks.  I missed that one.  Can you add one more action after the delete, add a couple of shard names, add a missing shard name, and other unit tests I requested?

Reply:
Ok. I'll add more actions after the delete.

Reply:
> add a couple of shard names

I have added other shards after the delete action

> add a missing shard name

This is only testing the addition and removal of metric labels `metrics.go`. It will not check whether label corresponds to an actual shard or not. That check is happening in the `controller.go`.

Reply:
Does it panic if you add a missing shard name?

Reply:
The `SetRouteMetricsControllerRoutesPerShardMetric` sets the label and the corresponding value for the metric. This function is being tested in the `metrics_test.go` unit test. It does not do any check whether the label corresponds to an actual shard or not. https://github.com/openshift/cluster-ingress-operator/pull/779/files#diff-2e4f9eb62dbb31f5371d6695c3fd4dffaf82ec1ee7b0a7d9cabaa1b2decd0abcR21-R23. It simply adds the label and the value. The check is happening in the `controller.go` file. So, in this unit test there would not be any concept of a missing shard or an actual available shard. Here, we're testing whether the added metrics is correctly being exported or not.